ext/openssl/openssl.c

```diff
@@ -2091,7 +2087,8 @@ PHP_FUNCTION(openssl_x509_parse)
 
 	cert = php_openssl_x509_from_param(cert_obj, cert_str);
 	if (cert == NULL) {
-		RETURN_FALSE;
+		zend_argument_type_error(1, "cannot be coerced into an X.509 Certificate");
+		RETURN_THROWS();
```

Comment:
I don't think any of these "cannot be coerced into an X.509 Certificate" should be converted. The certificate may well be invalid, and you may be using openssl_x509_parse to validate it...

Reply:
yeah agreed. that's good point!

Reply:
Are we just talking about the case where there is not a warning currently, or even those which have a warning?